### PR TITLE
SQL: fix object extraction from sources

### PR DESCRIPTION
**Pull request by imotov**
Throws an exception if hit extractor tries to retrieve unsupported
object. For example, selecting "a" from `{"a": {"b": "c"}}` now throws
an exception instead of returning null.

Relates to #37364

### CODE REVIEW

**Comment by imotov** [comment]
@elasticmachine run elasticsearch-ci-1

**Review by matriv — APPROVED** [review]
LGTM. nice catch! Thx!

**Comment by matriv** [comment]
run the gradle build tests 1

**Comment by matriv** [comment]
run the gradle build tests 2

**Comment by matriv** [comment]
run the gradle build tests 2

**Comment by matriv** [comment]
run the gradle build tests 1

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/execution/search/extractor/FieldHitExtractorTests.java [hunk 1]
```diff
@@ -336,6 +336,24 @@ public void testFieldWithDotsAndSamePathButDifferentHierarchy() {
         assertThat(ex.getMessage(), is("Multiple values (returned by [a.b.c.d.e.f.g]) are not supported"));
     }
 
+    public void testObjectsForSourceValue() throws IOException {
+        String fieldName = randomAlphaOfLength(5);
+        FieldHitExtractor fe = new FieldHitExtractor(fieldName, null, false);
+        SearchHit hit = new SearchHit(1);
```
Comment (matriv) [inline x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/execution/search/extractor/FieldHitExtractorTests.java]: Why did you choose to call `extract(hit)` and not follow the approach as here for example: https://github.com/elastic/elasticsearch/pull/37502/files#diff-9aaee9be08445653bba7407b9f2b5ca3R302

Comment (imotov) [reply x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/execution/search/extractor/FieldHitExtractorTests.java]: No particular reason. Both approaches would work I think. I can replace it if you prefer.

Comment (matriv) [reply x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/execution/search/extractor/FieldHitExtractorTests.java]: Nevermind, just wanted to check if there was a particular reason.

**Comment by imotov** [comment]
run the gradle build tests 1

**Comment by costin** [comment]
run the gradle build tests 1

**Comment by imotov** [comment]
run the gradle build tests 2

**Comment by imotov** [comment]
run the gradle build tests 1

**Comment by imotov** [comment]
run the gradle build tests 1

**Pull request merged**